tests/auto/resources/test_listentriesjob.cpp

```diff
@@ -0,0 +1,228 @@
+        QCOMPARE(&module, listEntries.module());
+    }
+
+    void shouldReturnCollection()
+    {
+        //GIVEN
+        Akonadi::Collection collection1;
+        collection1.setId(0);
+        Akonadi::Collection collection2;
+        collection2.setId(1);
+        SugarSession session(nullptr);
+        //WHEN
+        ListEntriesJob lej(collection1, &session);
+        //THEN
+        QVERIFY(lej.collection() == collection1);
+        QVERIFY(lej.collection() != collection2);
+    }
+
+    /*void changed()
+    {
+        //GIVEN
+        Akonadi::Collection c;
+
+        SugarSession s(nullptr);
+        ListEntriesJob lej(c, &s);
+        //WHEN
+
+        //THEN
+        QVERIFY(lej.collectionAttributesChanged());
+    }*/
+
+    void shouldReturnCorrectCount_data()
+    {
+        QTest::addColumn<QString>("moduleName");
+        QTest::addColumn<int>("correctCount");
+
+        QTest::newRow("Accounts") << "Accounts" << 3;
+        QTest::newRow("Opportunities") << "Opportunities" << 2;
+        QTest::newRow("Campaigns") << "Campaigns" << 1;
+    }
+
+    void shouldReturnCorrectCount()
+    {
+        QFETCH(QString, moduleName);
+        QFETCH(int, correctCount);
+        //GIVEN
+        Akonadi::Collection collection;
+        collection.setId(1);
+        SugarMockProtocol *protocol = new SugarMockProtocol;
+        SugarSession *session = new SugarSession (nullptr);
+        protocol->setSession(session);
+        session->setProtocol(protocol);
+        session->setSessionParameters("user", "password", "hosttest");
+        session->createSoapInterface();
+        ModuleHandler *handler;
+        if (moduleName == "Accounts") {
+            AccountsHandler *h = new AccountsHandler(session);
+            protocol->setAccountsHandler(h);
+            handler = h;
+        } else if (moduleName == "Opportunities") {
+            OpportunitiesHandler *h = new OpportunitiesHandler(session);
+            protocol->setOpportunitiesHandler(h);
+            handler = h;
+        } else if (moduleName == "Campaigns") {
+            CampaignsHandler *h = new CampaignsHandler(session);
+            protocol->setCampaignsHandler(h);
+            handler = h;
+        }
+        handler->initialCheck();
+        ListEntriesJob *job = new ListEntriesJob(collection, session);
+        job->setModule(handler);
+        QSignalSpy spy(job, SIGNAL(totalItems(int)));
+        //WHEN
+        job->start();
+        QVERIFY(job->exec());
+        //THEN
+        QCOMPARE(spy.count(), 1);
+        const QList<QVariant> arguments = spy.at(0);
+        QCOMPARE(arguments.at(0).toInt(), correctCount);
+
+        delete session;
+        delete handler;
+    }
+
+    void shouldReturnCorrectListAccounts()
+    {
+        SugarAccountCache *cache = SugarAccountCache::instance();
+        cache->clear();
+        //GIVEN
+        Akonadi::Collection collection;
+        collection.setId(1);
+        SugarMockProtocol *protocol = new SugarMockProtocol;
```

Comment:
doesn't seem to be deleted anywhere? (you could create it on the stack to avoid having to delete it manually)